### PR TITLE
SQL - Add Resource Identity swagger to common v2

### PR DESCRIPTION
**Pull request by viparek**
This PR has the fix for a breaking change that was caused due to a change related to ResourceIdentity in Server API.
The .net checks failure is expected and will be fixed in an ongoing PR(https://github.com/Azure/azure-powershell/pull/14997) which has a dependency on this PR.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [/] Other, please clarify - Updating an existing swagger to add resolve the naming issue that caused a breaking change. 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ /] SDK of .NET (need service team to ensure code readiness)
      - [ /] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ /] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ /] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ /] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [/ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/sql/resource-manager/common/v2/types.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          "format": "uuid",
+          "description": "The Azure Active Directory tenant id.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    }
+  },
+  "parameters": {
+    "SubscriptionIdParameter": {
```
Comment (jamestao) [inline specification/sql/resource-manager/common/v2/types.json:67]: You can remove the parameters section as well.

Comment (viparek) [reply specification/sql/resource-manager/common/v2/types.json]: Done

**Comment by jorgecotillo** [comment]
Make sure to always select all items that apply from the checklist.

**Review by jamestao — APPROVED** [review]
LGTM

**Review by zhenglaizhang — APPROVED** [review]

**Comment by mikhailshilkov** [comment]
I'm curious why this PR isn't marked as breaking. Effectively, it renames the type, which affects any typed downstream consumer.

I understand it's made to revert the previous breaking change but I'd expect it to be clearly marked and approved as breaking.

By the way, our native Pulumi Azure provider apparently ships changes faster than the Azure .NET SDK, so we already shipped the renamed types, and now this change is an actual break for us...

**Comment by ericshape** [comment]
> I'm curious why this PR isn't marked as breaking. Effectively, it renames the type, which affects any typed downstream consumer.
> 
> I understand it's made to revert the previous breaking change but I'd expect it to be clearly marked and approved as breaking.
> 
> By the way, our native Pulumi Azure provider apparently ships changes faster than the Azure .NET SDK, so we already shipped the renamed types, and now this change is an actual break for us...

@mikhailshilkov Thanks for your feedback! We appreciate it! 
We are sorry to break your pipeline due to the lack of a "breaking change" tag. We will be careful about it in the future. 

Here is why we did this breaking change:
In the recent Swagger release for 2020-11-01-preview API, we modified the Server and Managed Instance resources to add User Assigned Identity. To maintain backward compatibility with old APIs we introduced a new ResourceIdentityWithUserAssignedIdentities common entity which extended the old ResourceIdentities common entity. 
Unfortunately, on releasing the NET SDK we realized that a lot of customers who would be consuming the new SDK will break since the Object Class Name is changing. In order to address this issue, we remapped this new entity to a new version of ResourceIdentity to make sure the class name remains the same. This will prevent breaking a lot of customers which depend on the server/managed instance resources and they will now not need to make code changes to consume the new SDK.

**Comment by mikhailshilkov** [comment]
Thank you for the explanation and for labeling the issue as breaking.

By the way, I noticed that the "broken" name `ResourceIdentityWithUserAssignedIdentities` has already propagated into the Python SDK: https://github.com/Azure/azure-sdk-for-python/blob/890a9e4472a330d306d45c8e649042125ed16145/sdk/sql/azure-mgmt-sql/azure/mgmt/sql/models/_models.py#L11282

What is the plan there?

**Comment by ericshape** [comment]
> Thank you for the explanation and for labeling the issue as breaking.
> 
> By the way, I noticed that the "broken" name `ResourceIdentityWithUserAssignedIdentities` has already propagated into the Python SDK: https://github.com/Azure/azure-sdk-for-python/blob/890a9e4472a330d306d45c8e649042125ed16145/sdk/sql/azure-mgmt-sql/azure/mgmt/sql/models/_models.py#L11282
> 
> What is the plan there?

Yes. once we have done the .NET SDK release and Azure CLI release, we will request a new python SDK release next week. 
In case you don't know, our API/SDK/Clients have dependency:
.NET SDK is upstream of Powershell.
Python SDK is upstream of Azure CLI. 
```
                                                                 |--> .NET SDK ***  --> powershell ***
 generate swagger from SQL DB internal repo -> API specs repo ---
                                                                 |--> python SDK (auto) ---> Azure CLI***


*** manually, fix failed tests in corresponding repos, pass reviews
```

**Comment by ericshape** [comment]
> Thank you for the explanation and for labeling the issue as breaking.
> 
> By the way, I noticed that the "broken" name `ResourceIdentityWithUserAssignedIdentities` has already propagated into the Python SDK: https://github.com/Azure/azure-sdk-for-python/blob/890a9e4472a330d306d45c8e649042125ed16145/sdk/sql/azure-mgmt-sql/azure/mgmt/sql/models/_models.py#L11282
> 
> What is the plan there?

@mikhailalmeida 
We have had the latest Python SDK with the latest changes (include this PR). 
https://pypi.org/project/azure-mgmt-sql/0.29.0/#history
0.29.0 for track1

**Pull request merged**